src/include/storage/storage_structure/disk_array.h

```diff
@@ -231,6 +231,14 @@ class InMemDiskArray : public BaseInMemDiskArray<T> {
     InMemDiskArray(FileHandle& fileHandle, StorageStructureID storageStructureID,
         common::page_idx_t headerPageIdx, BufferManager* bufferManager, WAL* wal);
 
+    static inline common::page_idx_t initializeHeaderPage(BMFileHandle& fileHandle,
```

Comment:
This is fine but it seems like this function belongs to a utils class about disk_array, instead of disk_array. That's because it's not really used to modify a disk array, it's used to create a disk array in a file handle. If it's the only function it wouldn't matter but consider it.

Reply:
I think it's fine for now as this is the first static function here. We will extract them out later if we have more needs on more disk array static util functions.